Subject: [PATCH] Fix the Wi-Fi wrong status when turning Wi-Fi on

Previous status is still kept for a while when turning Wi-Fi on. This
is because when just turning Wi-Fi on, some fields of AccessPoint are
still old.

This patch removes null check to set null value to them. This results
in reflecting the latest status.

diff --git a/packages/SettingsLib/src/com/android/settingslib/wifi/WifiTracker.java b/packages/SettingsLib/src/com/android/settingslib/wifi/WifiTracker.java
index 0a32f19..b30fb42 100644
--- a/packages/SettingsLib/src/com/android/settingslib/wifi/WifiTracker.java
+++ b/packages/SettingsLib/src/com/android/settingslib/wifi/WifiTracker.java
@@ -347,9 +347,7 @@
                     continue;
                 }
                 AccessPoint accessPoint = getCachedOrCreate(config, cachedAccessPoints);
-                if (mLastInfo != null && mLastNetworkInfo != null) {
-                    accessPoint.update(connectionConfig, mLastInfo, mLastNetworkInfo);
-                }
+                accessPoint.update(connectionConfig, mLastInfo, mLastNetworkInfo);
                 if (mIncludeSaved) {
                     // If saved network not present in scan result then set its Rssi to MAX_VALUE
                     boolean apFound = false;
